library/pkparse.c

```diff
@@ -683,7 +683,7 @@ static int pk_parse_key_rfc8410_der(mbedtls_pk_context *pk,
 }
 #endif /* MBEDTLS_PK_HAVE_RFC8410_CURVES */
 
-#if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
+#if defined(MBEDTLS_PK_USE_PSA_EC_DATA) && defined(MBEDTLS_ECP_LIGHT)
```

Comment:
Note: as discussed, #7442 should change that to `MBEDTLS_PK_PARSE_EC_COMPRESSED`. (Nothing to change here, just for reference.)